dev/src/lobster/geom.h

```diff
@@ -242,6 +242,65 @@ template<typename T, int N> struct vec : basevec<T, N> {
     template<typename T2, int C, int R> friend class matrix;
 };
 
+template<typename T, int N> struct vec_iterator {
+    vec<T, N> min;
+    vec<T, N> max;
+    vec<T, N> value;
+
+    vec_iterator(vec<T, N> min, vec<T, N> max) : min(min), max(max), value(min) {}
+    vec_iterator(vec<T, N> min, vec<T, N> max, vec<T, N> value)
+        : min(min), max(max), value(value) {}
+
+    vec<T, N> &operator*() {
+        return value;
+    }
+    vec<T, N> *operator->() {
+        return &value;
+    }
+
+    vec_iterator &operator++() {
+        for (int i = 0; i < N; ++i) {
```

Comment:
This iterator is certainly neat, and avoids the usual 3 nested for loops I have a bunch of thruout the code. But I'd be warey to use it in any case where the actual speed of the iteration is likely a factor in the overall speed (very small body of the innerloop), since even though this `operator++` typically does not go thru the `for` more than once, its still more complicated logic the compiler can't completely get out of the way of the innermost loop. Probably fine in most cases though.